src/contracts/Staking.sol

```diff
@@ -515,7 +527,7 @@ contract Staking is Ownable {
         }
         _retrieveBalanceFromUser(_amount, msg.sender);
 
-        Claim memory userCoolInfo = coolDownInfo[msg.sender];
+        Claim storage userCoolInfo = coolDownInfo[msg.sender];
```

Comment:
you were right on this one, I didn't think about the value being mutated from claimWithdraw.  Added tests surrounding this